backend/src/router.go

```diff
@@ -176,6 +177,77 @@ func getTutorsUsername(c *gin.Context) {
 	})
 }
 
+
+// Handler for PUT /tutors/:username. Returns an empty json if sucessful. 
+// Errors if:
+//
+//  - There is no authenticated user (401)
+//  - A server issue prevents parsing the JWT (500)
+//  - Path username is different than logged in user (404)
+//  - The user does not exist (404)
+//  - The user is already a tutor (404)
+//
+// Response Schema: {
+//   200
+// }
+// Error Schema: {
+//   error: String
+// }
+
+func putTutorsUsername(c *gin.Context) {
```

Comment:
Looking at the [CRUD docs](https://docs.microsoft.com/en-us/iis-administration/api/crud) again, I think using a `POST /tutors` route is better for this. `POST` is used for create while `PUT` is for update. Using `POST` is more accurate for CRUD (`PUT` should be idempotent anyways and we're asserting the user already exists). Switching to just `/tutors` means we also don't have to check if usernames match since it pulls from the authenticated user.

Reply:
Switched it over. Seems to work. Pushing soon.